### ISSUE TITLE
GridTE performance

### ISSUE THREAD

**Issue by Sir-Will** [McJtyMods/RFToolsPower#7]
Any chance this can be improved?
![](https://i.imgur.com/1llZazK.jpg)

**Comment by McJty** [comment]
How many grids do you have on the multiblock?

**Comment by Sir-Will** [comment]
Just noticed that I posted this in the wrong repo, sorry about that.
Want me to move it to EFab?

**Comment by McJty** [comment]
It's ok. Most likely the issue is multiple grids on the same efab multiblock and that issue is already fixed and ready for next release

**Comment by Sir-Will** [comment]
Is that commit already on github?

**Comment by Sir-Will** [comment]
The clients don't need to be updated as the grid calculation is only done by the server, except you do something out of the ordinary.
This also shouldn't be complicated if the fix is not a complete rewrite of multiple classes.

**Comment by McJty** [comment]
No but a lot of other things have changed in the mod too. You could try to get an older version and try to do the same fix on that.  It is not a complicated fix

**Comment by Sir-Will** [comment]
That's why I'm was asking about the commit to see what was changed to fix this issue.
It's at 9% now so I really would like to get this issue fixed asap and also test if it actually fixes this issue.
![](https://i.imgur.com/uS8whl9.jpg)

**Comment by Sir-Will** [comment]
Is it this fix you were talking about? https://github.com/McJtyMods/EFab/commit/e32edde29bc067688a5b572d6cb1a5dc42c26044

**Comment by McJty** [comment]
yes indeed. Note that yesterday I released alpha versions of all my mods. But maybe that's too risky and complicated to distribute to all your uses. In any case, the commit above is the fix you want

**Comment by Sir-Will** [comment]
Hm, that removed code doesn't use func_180495_p (getBlockState) which is the main issue.

**Comment by McJty** [comment]
It does. Indirectly. I'm pretty sure that will be the reason. The removed code makes sure that there is no cascade of updates which all use getBlockState
